runtime/autoload/tar.vim

```diff
@@ -432,11 +433,14 @@ if v:shell_error != 0
   set nomod
 
   let &report= repkeep
+  exe "cd ".pwdkeep
+  silent exe "file tarfile::".escape_file
 endfun
 
 " ---------------------------------------------------------------------
 " tar#Write: {{{2
 fun! tar#Write(fname)
+  let pwdkeep = getcwd()
```

Comment:
@habamax while in `tar#Read` we can use `curdir` because is set to the original working directory, in `tar#Write` it seems to be set to a temporary directory instead, e.g. `/tmp/vemgadL/1` (tried this in Linux Alpine), so we probably still need a separate variable to restore the original pwd?

Reply:
I believe it's because in `tar#Read` https://github.com/vim/vim/blob/fa32c721d73b53e5501e4b2d6d4fba47334a9a12/runtime/autoload/tar.vim#L301 `b:curdir` is set to the temporary directory, and in `tar#Write` https://github.com/vim/vim/blob/fa32c721d73b53e5501e4b2d6d4fba47334a9a12/runtime/autoload/tar.vim#L447 `curdir` is set to `b:curdir` that still points to the temporary directory.

Reply:
So while it might not be DRY, having a dedicated variable, like `pwdkeep`, at least in `tar#Write`, seems the easiest way to restore the original pwd, and at that stage might as well use it in `tar#Read` for consistency?

Reply:
For tar#Write for sure we need another var to keep&restore current directory.

It is not clear what problem OG author had, but apparently there was something.

Reply:
I wonder if there are test cases, simple grep for `tar#Write` shows nothing :(

Reply:
What the original author meant to do goes beyond my understanding. I'll keep the `pwdkeep` variable for `tar#Write` while implementing the other suggested changes.